Demo/app/src/main/java/com/juanocampo/swipeable/swipeablelist/swpeable/adapter/SwipeableHelperAdapter.java

```diff
@@ -17,40 +20,39 @@
 
 public abstract class SwipeableHelperAdapter extends RecyclerView.Adapter {
 
+
     public interface SwipeAdapterActions {
         void swiped(int position);
     }
 
     private final SwipeableHelperCallback callback;
     private SwipeAdapterActions listener;
     private List<RecyclerView.ViewHolder> swipeableViewHolders;
+    private final LinearLayoutManager manager;
 
 
-    public SwipeableHelperAdapter(SwipeAdapterActions listener, RecyclerView recyclerView) {
-        callback = new SwipeableHelperCallback(this);
-        ItemTouchHelper mItemTouchHelper = new ItemTouchHelper(callback);
-        mItemTouchHelper.attachToRecyclerView(recyclerView);
+    public SwipeableHelperAdapter(SwipeAdapterActions listener, RecyclerView recyclerView, LinearLayoutManager manager) {
+        callback = new SwipeableHelperCallback(this, recyclerView);
         this.listener = listener;
         this.swipeableViewHolders = new ArrayList<>();
+        this.manager = manager;
+        attachToRecycler(recyclerView);
+    }
+
+    public void attachToRecycler(RecyclerView recyclerView) {
+        ItemTouchHelper mItemTouchHelper = new ItemTouchHelper(callback);
+        mItemTouchHelper.attachToRecyclerView(recyclerView);
     }
 
+
+
     @Override
     public final RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         RecyclerView.ViewHolder itemViewHolder = onCreateSwipeViewHolder(parent, viewType);
         swipeableViewHolders.add(itemViewHolder);
```

Comment:
this is not the best approach because as the list grows, this list is going to get bigger and bigger. 
And if the list decides to re-use the viewHolder, you will end up animating the wrong item.